src/graphchain/core.py

```diff
@@ -242,7 +254,7 @@ def time_to_result(self, memoize: bool = True) -> float:
     @property
     def cache_filename(self) -> str:
         """Filename of the cache file to load or store."""
-        return f"{self.hash}.joblib.lz4"
+        return f"{self.hash}.dat"
```

Comment:
It might be good if this were customizable. For instance if I save to parquet it's because I want to be able to interrogate it with parquet reading tools, which will expect the `.parquet` extension.

Reply:
For instance, you could add a new function to the optimizer which is a callable that returns the file extension for any given file.

Reply:
Done. The serialize/deserialize function is now in control over the filename written to the filesystem.